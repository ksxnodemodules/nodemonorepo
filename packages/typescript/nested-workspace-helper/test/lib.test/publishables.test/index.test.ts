import createSetupTeardown from '../../.lib/setup-teardown'
import {classifyPublishability} from '../../../index'

import {
  PublishablePackageList,
  UnpublisablePackageList,
  PrivatePackageList,
  PublishableClassification
} from '../../../lib/types'

type PackageList = PublishablePackageList | UnpublisablePackageList | PrivatePackageList

const {apply} = createSetupTeardown('publishability.yaml')

describe('matches snapshot', () => {
  it('when there are all 3 types', apply(async () => {
    const classified = await classifyPublishability('root')
    expect(classified).toMatchSnapshot()
  }))

  it('when there is only publishable packages', apply(async () => {
    const classified = await classifyPublishability('root/toBePublishable')
    expect(classified).toMatchSnapshot()
  }))

  it('when there is only unpublishable packages', apply(async () => {
    const classified = await classifyPublishability('root/toBeUnpublishable')
    expect(classified).toMatchSnapshot()
  }))

  it('when there is only skipped packages', apply(async () => {
    const classified = await classifyPublishability('root/toBeSkipped')
    expect(classified).toMatchSnapshot()
  }))
})
