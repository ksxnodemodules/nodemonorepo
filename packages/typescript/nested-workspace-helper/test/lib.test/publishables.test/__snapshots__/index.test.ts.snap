// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when there are all 3 types 1`] = `
Object {
  "publishables": Array [
    Object {
      "manifestContent": Object {
        "__describe": "Package has yet to exist in registry",
        "__expect": "Should be publishable",
        "name": "to-be-new",
        "version": "0.0.0",
      },
      "manifestFile": "root/toBePublishable/toBeNew/package.json",
      "path": "root/toBePublishable/toBeNew",
    },
    Object {
      "manifestContent": Object {
        "__describe": "Package's version (10.11.12) does not exist in registry",
        "__expect": "Should be publishable",
        "name": "to-not-be-duplicated",
        "version": "10.11.12",
      },
      "manifestFile": "root/toBePublishable/toNotBeDuplicated/package.json",
      "path": "root/toBePublishable/toNotBeDuplicated",
    },
  ],
  "skip": Array [
    Object {
      "manifestContent": Object {
        "__describe": "Private package",
        "__expect": "Should be in skip",
        "name": "to-be-private",
        "private": true,
        "version": "0.0.0",
      },
      "manifestFile": "root/toBeSkipped/toBePrivate/package.json",
      "path": "root/toBeSkipped/toBePrivate",
    },
    Object {
      "manifestContent": Object {
        "__describe": "Package has no name",
        "__expect": "Should be in skip",
        "version": "0.0.0",
      },
      "manifestFile": "root/toBeSkipped/toNotBeNamed/package.json",
      "path": "root/toBeSkipped/toNotBeNamed",
    },
  ],
  "unpublishables": Array [
    Object {
      "manifestContent": Object {
        "__describe": "Package's version (1.2.3) already exist in registry",
        "__expect": "Should be unpublishable",
        "name": "to-be-duplicated",
        "version": "1.2.3",
      },
      "manifestFile": "root/toBeUnpublishable/toBeDuplicated/package.json",
      "path": "root/toBeUnpublishable/toBeDuplicated",
    },
    Object {
      "manifestContent": Object {
        "__describe": "Package does not have a version",
        "__expect": "Should be unpublishable",
        "name": "to-not-have-version",
      },
      "manifestFile": "root/toBeUnpublishable/toNotHaveVersion/package.json",
      "path": "root/toBeUnpublishable/toNotHaveVersion",
    },
  ],
}
`;

exports[`matches snapshot when there is only publishable packages 1`] = `
Object {
  "publishables": Array [
    Object {
      "manifestContent": Object {
        "__describe": "Package has yet to exist in registry",
        "__expect": "Should be publishable",
        "name": "to-be-new",
        "version": "0.0.0",
      },
      "manifestFile": "root/toBePublishable/toBeNew/package.json",
      "path": "root/toBePublishable/toBeNew",
    },
    Object {
      "manifestContent": Object {
        "__describe": "Package's version (10.11.12) does not exist in registry",
        "__expect": "Should be publishable",
        "name": "to-not-be-duplicated",
        "version": "10.11.12",
      },
      "manifestFile": "root/toBePublishable/toNotBeDuplicated/package.json",
      "path": "root/toBePublishable/toNotBeDuplicated",
    },
  ],
  "skip": Array [],
  "unpublishables": Array [],
}
`;

exports[`matches snapshot when there is only skipped packages 1`] = `
Object {
  "publishables": Array [],
  "skip": Array [
    Object {
      "manifestContent": Object {
        "__describe": "Private package",
        "__expect": "Should be in skip",
        "name": "to-be-private",
        "private": true,
        "version": "0.0.0",
      },
      "manifestFile": "root/toBeSkipped/toBePrivate/package.json",
      "path": "root/toBeSkipped/toBePrivate",
    },
    Object {
      "manifestContent": Object {
        "__describe": "Package has no name",
        "__expect": "Should be in skip",
        "version": "0.0.0",
      },
      "manifestFile": "root/toBeSkipped/toNotBeNamed/package.json",
      "path": "root/toBeSkipped/toNotBeNamed",
    },
  ],
  "unpublishables": Array [],
}
`;

exports[`matches snapshot when there is only unpublishable packages 1`] = `
Object {
  "publishables": Array [],
  "skip": Array [],
  "unpublishables": Array [
    Object {
      "manifestContent": Object {
        "__describe": "Package's version (1.2.3) already exist in registry",
        "__expect": "Should be unpublishable",
        "name": "to-be-duplicated",
        "version": "1.2.3",
      },
      "manifestFile": "root/toBeUnpublishable/toBeDuplicated/package.json",
      "path": "root/toBeUnpublishable/toBeDuplicated",
    },
    Object {
      "manifestContent": Object {
        "__describe": "Package does not have a version",
        "__expect": "Should be unpublishable",
        "name": "to-not-have-version",
      },
      "manifestFile": "root/toBeUnpublishable/toNotHaveVersion/package.json",
      "path": "root/toBeUnpublishable/toNotHaveVersion",
    },
  ],
}
`;
